vue/sbc-common-components/src/components/SbcHeader.vue

```diff
@@ -590,8 +592,17 @@ export default class SbcHeader extends Mixins(NavigationMixin) {
     await this.syncCurrentAccount(settings)
     this.$emit('account-switch-completed')
 
+    // check for allowed redirect to determine whether need to redirect back to that page or dashboard
+    // list of Allowed URLs
+    const allowedRedirectURls = ConfigHelper.getAllowedUrlForRedirectToSamePage()
+    // default redirect to dashboard
+    const defaultRedirect = `${ConfigHelper.getRegistryHomeURL()}/${Pages.REGISTRY_DASHBOARD}`
+
+    const currentURL = trimTrailingSlashURL(`${window.location.origin}${window.location.pathname}`)
+    const redirect = allowedRedirectURls.indexOf(currentURL) > -1 ? currentURL : defaultRedirect
+
     if (!inAuth) {
```

Comment:
I think for the future it would be nice if we allowed another prop condition here. 

Maybe not `inAuth`, but something like `skipAccountSwitchRedirect` (default false)? 

That way in the future various applications could handle the 
`'account-switch-completed'` event if they needed and bypass the re-direct?

Reply:
commented in code now, In future when someone wants to use, they have made changes in applications also, Then we can add this prop. Now if we gave it as a prop i think it will skip all the checks like NFS, Account pending